pkg/plugin/stream_reader.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2017 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"io"
+)
+
+// ReceiveFunc is a function that either returns a slice
+// of an arbitrary number of bytes OR an error. Returning
+// an io.EOF means there is no more data to be read; any
+// other error is considered an actual error.
+type ReceiveFunc func() ([]byte, error)
```

Comment:
Any thoughts on my previous comment about using an interface instead of a function pointer?

Reply:
I'm fine with that -- I originally went with funcs because I didn't want the interface def to be tied just to a grpc Bytes stream client, and I didn't want to define another adapter type, but that's all it's used for. Are you thinking:

```
type ByteReceiver interface {
    Recv() (*generated.Bytes, error)
    CloseSend() error
}
``` 
that the Bytes stream client already satisfies? or a more generic
```
type ByteReceiver interface {
    Receive() ([]byte, error)
    Close() error
}
```
along with an adapter?

Reply:
Ah, I see the former is exactly what you commented before. I'm happy with that.